pylabrobot/visualizer/lib.js

```diff
@@ -701,6 +719,47 @@ class Trough extends Container {
   }
 }
 
+class Reservoir extends Resource {
+  constructor(resourceData, parent = undefined) {
+    super(resourceData, parent);
+    const { num_items_x, num_items_y } = resourceData;
+    this.num_items_x = num_items_x;
+    this.num_items_y = num_items_y;
+  }
+
+  drawMainShape() {
+    return new Konva.Rect({
+      width: this.size_x,
+      height: this.size_y,
+      fill: "#2B2D42",
+      stroke: "black",
+      strokeWidth: 1,
+    });
+  }
+
+  serialize() {
+    return {
+      ...super.serialize(),
+      ...{
+        num_items_x: this.num_items_x,
+        num_items_y: this.num_items_y,
+      },
+    };
+  }
+
+  update() {
+    super.update();
+
+    // Rename the children
+    for (let i = 0; i < this.num_items_x; i++) {
+      for (let j = 0; j < this.num_items_y; j++) {
+        const child = this.children[i * this.num_items_y + j];
+        child.name = `${this.name}_well_${i}_${j}`;
+      }
+    }
+  }
+}
```

Comment:
is the reservoir js class still needed? it seems to be a relic from this: https://github.com/PyLabRobot/pylabrobot/pull/149

Reply:
Nope! That should be deleted.

Reply:
but the OT reservoirs are just `Plate`s in PLR

Reply:
Yeah, I just had them super the methods from plate in the visualizer. I think keeping them as their own class is better because it allows for more utility features if, for some reason in the future, someone wants specific utility with reservoirs.

Reply:
@rickwierenga Is it okay to have the visualizer super the `Plate` methods?

Reply:
the way visualizer classes are found is through `classForResourceType`, which looks at `resourceData.type`. In PLR, the type is never `Reservoir` because no class `Reservoir` exists in Python. I think we should add this back if we ever add a `Reservoir` class in python plr.